### PR TITLE
feat: use app settings for file upload validations #154

### LINKED ISSUE #154: Use image/file upload settings received from backend
`client.getAppSettings()`

- [x] Load app settings on init
- [x] Propagate app settings via Observable
- [X] Add client side checks for file uploads (extension and mime type)
- [x] Add documentation

### PR DESCRIPTION
**Pull request by szuperaz**
closes #154

### CODE REVIEW

File: projects/stream-chat-angular/src/lib/chat-client.service.ts [hunk 1]
```diff
@@ -39,6 +42,8 @@ export class ChatClientService {
         `stream-chat-angular-${version}-${this.chatClient.getUserAgent()}`
       );
     });
+    const settings = await this.chatClient.getAppSettings();
```
Comment (petyosi) [inline projects/stream-chat-angular/src/lib/chat-client.service.ts:45]: This is a relatively expensive backend call, which we don't want to do for every client - many users won't upload files during their sessions. Can this be done on demand?

**Review by petyosi — CHANGES_REQUESTED** [review]
The getAppSettings should be done just-in-time and cached for subsequent calls.

File: projects/stream-chat-angular/src/lib/chat-client.service.ts (hunk 1 above)
Comment (szuperaz) [reply projects/stream-chat-angular/src/lib/chat-client.service.ts:45]: I can shift things around and only load these settings when the user tries to upload files

Comment (szuperaz) [reply projects/stream-chat-angular/src/lib/chat-client.service.ts]: Done

**Review by petyosi — APPROVED** [review]

**Pull request merged**